// node-opcua-nodeset-di/dist/enum_device_health.js
"use strict";
// ----- this file has been automatically generated - do not edit
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnumDeviceHealth = void 0;
/**
 * |           |                                                  |
 * |-----------|--------------------------------------------------|
 * | namespace |http://opcfoundation.org/UA/DI/                   |
 * | nodeClass |DataType                                          |
 * | name      |1:DeviceHealthEnumeration                         |
 * | isAbstract|false                                             |
 */
var EnumDeviceHealth;
(function (EnumDeviceHealth) {
    /**
     * This device functions normally.
     */
    EnumDeviceHealth[EnumDeviceHealth["NORMAL"] = 0] = "NORMAL";
    /**
     * Malfunction of the device or any of its
     * peripherals.
     */
    EnumDeviceHealth[EnumDeviceHealth["FAILURE"] = 1] = "FAILURE";
    /**
     * Functional checks are currently performed.
     */
    EnumDeviceHealth[EnumDeviceHealth["CHECK_FUNCTION"] = 2] = "CHECK_FUNCTION";
    /**
     * The device is currently working outside of its
     * specified range or that internal diagnoses
     * indicate deviations from measured or set values.
     */
    EnumDeviceHealth[EnumDeviceHealth["OFF_SPEC"] = 3] = "OFF_SPEC";
    /**
     * This element is working, but a maintenance
     * operation is required.
     */
    EnumDeviceHealth[EnumDeviceHealth["MAINTENANCE_REQUIRED"] = 4] = "MAINTENANCE_REQUIRED";
})(EnumDeviceHealth = exports.EnumDeviceHealth || (exports.EnumDeviceHealth = {}));
